yt/frontends/adaptahop/data_structures.py

```diff
@@ -0,0 +1,288 @@
+"""
+Data structures for AdaptaHOP frontend.
+
+
+
+
+"""
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2013, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+import numpy as np
+import stat
+import os
+import re
+
+from yt.data_objects.data_containers import \
+    YTSelectionContainer
+from yt.data_objects.static_output import \
+    Dataset
+from yt.frontends.halo_catalog.data_structures import \
+    HaloCatalogFile
+from yt.funcs import \
+    setdefaultattr
+from yt.geometry.particle_geometry_handler import \
+    ParticleIndex
+from yt.utilities.cython_fortran_utils import FortranFile
+from yt.units import Mpc
+
+from .definitions import \
+    HEADER_ATTRIBUTES
+from .fields import \
+    AdaptaHOPFieldInfo
+
+class AdaptaHOPBinaryFile(HaloCatalogFile):
+    def __init__(self, ds, io, filename, file_id):
+        super(AdaptaHOPBinaryFile, self).__init__(ds, io, filename, file_id)
+
+    def _read_particle_positions(self, ptype, f=None):
+        raise NotImplementedError
+
+class AdaptaHOPDataset(Dataset):
+    _index_class = ParticleIndex
+    _file_class = AdaptaHOPBinaryFile
+    _field_info_class = AdaptaHOPFieldInfo
+
+    _code_length_to_Mpc = Mpc.to('cm').value / 3.08e24
```

Comment:
Is this because there's a setting in the underlying code that assumes a megaparsec is 3.08e24 cm?  If so, this looks good (could use a comment explaining) but otherwise I'd prefer it to use the underlying values from the units system.